index.js

```diff
@@ -54,13 +56,76 @@ function decodeAssetCacheName(encodedString) {
   return decodedFileName.split(":");
 }
 
+function touch(path) {
+  const time = new Date();
+  try {
+    fs.utimesSync(path, time, time);
+  } catch (err) {
+    fs.closeSync(fs.openSync(path, "w"));
+  }
+}
+
+function evictLeastRecentlyUsed(cacheDir, maxSize, logger) {
+  getFolderSize(cacheDir, (err, size) => {
+    if (err) {
+      throw err;
+    }
+
+    if (size >= maxSize) {
+      try {
+        // find least recently used file
+        const leastRecentlyUsed = findLeastRecentlyUsed(cacheDir);
+
+        // and delete it
+        const { dir } = path.parse(leastRecentlyUsed.path);
+        fs.unlinkSync(leastRecentlyUsed.path);
+        fs.rmdirSync(dir);
+
+        if (logger) {
+          logger.info(`Evicted ${leastRecentlyUsed.path} from cache`);
+        }
+
+        evictLeastRecentlyUsed(cacheDir, maxSize, logger);
+      } catch (e) {
+        logger.error(e);
+      }
+    }
+  });
+}
+
+function findLeastRecentlyUsed(dir, result) {
```

Comment:
all the `fs` calls in this function should probably also be wrapped in `try/catch` blocks. and if those fail, they should also be logged.

Reply:
hm... I thought so too, then I thought, this function is only going to be used inside `evictLeastRecentlyUsed`, so an occasional exception can be caught there?

Reply:
In this case I would argue that it's probably better to define `findLeastRecentlyUsed` as a `const` right above `const leastRecentlyUsed = findLeastRecentlyUsed(cacheDir);`. The scope matches the functions usage and you can loose the comment since its self documenting.

Reply:
I see what you mean... the problem is, to make it accessible for testing, I need to attach it to the `module.exports` ... it's not really the most beautiful solution, and affords refactoring everything into more constrained small modules, but for now...

Reply:
I see - in this case this a probably a pragmatic tradeoff. 👍